src/udp_socket.cpp

```diff
@@ -311,6 +311,16 @@ void udp_socket::on_read(error_code const& ec, udp::socket* s)
 
 		if (err == boost::asio::error::would_block || err == boost::asio::error::try_again) break;
 		on_read_impl(ep, err, bytes_transferred);
+
+		// found on iOS, socket will be disconnected when app goes backgroud. try to reopen it.
+		if (err == boost::asio::error::not_connected) 
+		{
+			ep = s->local_endpoint(err);
+			if (!err) {
```

Comment:
in fact, in the case where you call bind, you probably don't want to continue either. I think you should replace that break with a return;